lib/ansible/executor/task_executor.py

```diff
@@ -542,7 +542,15 @@ def _execute(self, variables=None):
         # get the connection and the handler for this execution
         if (not self._connection or
                 not getattr(self._connection, 'connected', False) or
-                not self._connection.matches_name([current_connection]) or
+                not (
+                    (load_info := self._shared_loader_obj.connection_loader.get_with_context(
+                        current_connection,
+                        self._play_context,
```

Comment:
If we can switch back to just a `find_plugin_with_context` instead of instantiating, all these args can also go away.